pkg/parse/json/json_test.go

```diff
@@ -0,0 +1,181 @@
+package json_test
+
+import (
+	"bytes"
+	gojson "encoding/json"
+	"io/ioutil"
+	"fmt"
+	"github.com/tektoncd/triggers/pkg/parse"
+	. "github.com/tektoncd/triggers/pkg/parse/json"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func TestFieldsFromByteMessage(t *testing.T) {
+	var testNames []string
+	var payloads [][]byte
+	// Populates payloads using examples
+	err := filepath.Walk("./examples", func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			t.Errorf("Failure accessing path %q: %v\n", path, err)
+			return err
+		}
+		if !info.IsDir() {
+			t.Logf("Reading %s",path)
+			b, err := ioutil.ReadFile(path)
+			if err != nil {
+				return err
+			}
+			payloads = append(payloads,b)
+			fileNameTrimmed := strings.TrimSuffix(path,".json")
+			testNames = append(testNames, strings.Replace(strings.Title(fileNameTrimmed),"/","",-1))
+		}
+		return nil
+	})
+	if err != nil {
+		t.Fatalf("Unable to load example payloads: %v",err)
+	}
```

Comment:
i'd move grabbing all these files to a separate function (and probably calling t.Helper inside it!)